src/Browscap/Data/Validator/DivisionDataValidator.php

```diff
@@ -0,0 +1,400 @@
+<?php
+declare(strict_types = 1);
+namespace Browscap\Data\Validator;
+
+use Assert\Assertion;
+
+class DivisionDataValidator implements ValidatorInterface
+{
+    /**
+     * valdates the structure of a division
+     *
+     * @param array  $divisionData  Data to validate
+     * @param string $filename
+     * @param array  &$allDivisions
+     * @param bool   $isCore
+     */
+    public function validate(
+        array $divisionData,
+        string $filename,
+        array &$allDivisions = [],
```

Comment:
What's this by-ref used for? Definitely a code smell here; can we avoid...? Especially as the return type is `void`...

Reply:
This is required to detect duplicate divisions.

Reply:
Urgh, ok